File: OEP/Kodgyak/Ora/02.23/gyak02/gyak02/Geometry_skel/Geometry/Point.cs
using System;

namespace Geometry;

public class Point
{
    private double x;
    private double y;

    //ctor / ctorf 
    public Point(double a, double b)
    {
        x = a;
        y = b;
    }

    public double Distance(Point p)
    {
        return Math.Sqrt((x - p.x) * (x - p.x) + (y - p.y) * (y - p.y));
    }
}

File: Prognyelvek (Java)/Gyakorlat/_02.27_Csomagok/point2d/Point.java
package point2d;

public class Point{ //DB7ZTC

    public double x = 0.0;
    public double y = 0.0;

    public Point (){
        this.x=0.0;
        this.y=0.0;
    }

    public void move(double dx, double dy){
        this.x+=dx;
        this.y+=dy;
    }
    
    public void mirror(Point p){
        this.x=p.x-(x-p.x); //2*cx - x;
        this.y=p.y-(y-p.y);
    }

    public void printPoint(){
        System.out.println("Pont koordinatai: ("+x+","+y+")");
    }

    public double distance( Point p){
        return Math.sqrt(Math.pow(this.x-p.x,2)+Math.pow(this.y-p.y,2));
    }
}